jitsi-utils-kotlin/src/main/kotlin/org/jitsi/utils/queue/QueueStatistics.kt

```diff
@@ -0,0 +1,227 @@
+/*
+* Copyright @ 2018 - present 8x8, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.jitsi.utils.queue
+
+import org.jitsi.utils.OrderedJsonObject
+import org.jitsi.utils.stats.BucketStats
+import org.jitsi.utils.stats.RateStatistics
+import org.json.simple.JSONObject
+import java.util.Collections
+import java.util.concurrent.atomic.LongAdder
+import java.util.IdentityHashMap
+import java.util.concurrent.ConcurrentHashMap
+
+class QueueStatisticsInternal(queueSize: Int) {
```

Comment:
Probably just confused by the diff, but I'm a little confused by `QueueStatisticsInternal` vs `QueueStatistics`

Reply:
`QueueStatisticsInternal` is the object that's instantiated as `localStats` and `globalStats`, whereas `QueueStatistics` is the object that actually observes the queue, calculates the current size and (possibly) linger time, and then updates both statistics objects.  Better names welcome....

Reply:
`QueueStatisticsInternal` -> `QueueStatistics`
`QueueStatistics` -> `QueueObserver` ?